core/src/ipld.rs

```diff
@@ -159,6 +159,38 @@ impl Ipld {
     }
 }
 
+#[cfg(feature = "quickcheck")]
+impl quickcheck::Arbitrary for Ipld {
+    fn arbitrary(g: &mut quickcheck::Gen) -> Self {
+        let bool_item = bool::arbitrary(g);
+        let int_item = i128::arbitrary(g);
+        let float_item = f64::arbitrary(g);
+        let string_item = String::arbitrary(g);
+        let bytes_item = Vec::arbitrary(g);
+        let list_item = Vec::arbitrary(g);
+        let map_item = BTreeMap::arbitrary(g);
```

Comment:
In my experience you’ll also need to place a bound on the recursion happening here.

Reply:
Fixed